In [46]:
import numpy as np
import pandas as pd
import plotly.express as px
import matplotlib.pyplot as plt

In [47]:
eat = pd.read_excel("SharkData.xlsx", sheet_name = 0)
targets = pd.read_excel("SharkData.xlsx", sheet_name = 1)
drops = pd.read_excel("SharkData.xlsx", sheet_name = 2)
other_factors = pd.read_excel("SharkData.xlsx", sheet_name = 3)

Cleaning for eat Dataframe

In [48]:
eat = eat.iloc[1:,:-1]
eat.rename(columns = {"Unnamed: 14":'Etc. Comments', "Pieces Eaten": "Date"}, inplace = True)

#get rid of null values at end of dataset
eat = eat.head(582)

# Change date to just date format, not datetime
eat['Date'] = pd.to_datetime(eat['Date']).dt.date

eat.tail()

Date  Ross  Chandler  BT1  BT2  BT3  BT4  BT5  GR1  GR2  GR3  GR4  \
578  2020-09-22   6.0       0.0  1.0  2.0  6.0  0.0  4.0  1.0  0.0  0.0  5.0   
579  2020-09-24   0.0       9.0  0.0  7.0  5.0  5.0  7.0  0.0  1.0  2.0  2.0   
580  2020-09-26   NaN       NaN  NaN  NaN  NaN  NaN  NaN  4.0  0.0  2.0  3.0   
581  2020-09-27   5.0       6.0  3.0  6.0  4.0  6.0  3.0  NaN  NaN  NaN  NaN   
582  2020-09-29  10.0       3.0  1.0  1.0  4.0  2.0  7.0  4.0  4.0  0.0  2.0   

     GR5  Total:  Etc. Comments  
578  1.0     26.0           NaN  
579  6.0     44.0           NaN  
580  3.0     12.0           NaN  
581  NaN     33.0           NaN  
582  0.0     38.0           NaN

# 
Look at means of Individual sharks, then look at means of individual species

In [74]:
# Individual mean number of pieces eaten
All_Sharks = eat[['Ross', 'Chandler','BT1','BT2','BT3','BT4','BT5','GR1','GR2','GR3','GR4', 'GR5']]
All_mean = round(All_Sharks.mean(), 3)
All_mean

Ross        3.552
Chandler    2.882
BT1         2.043
BT2         2.858
BT3         2.349
BT4         2.790
BT5         3.076
GR1         1.346
GR2         1.352
GR3         1.858
GR4         1.602
GR5         1.735
dtype: float64

In [75]:
# Graph Mean Number of Fish Eaten For Each Shark

Eaten_Means = {'Shark': ['Ross', 'Chandler','BT1','BT2','BT3','BT4','BT5','GR1','GR2','GR3','GR4', 'GR5'],
        'Mean': [3.552,2.882,2.043,2.858,2.349,2.790,3.076,1.346,1.352,1.858,1.602,1.735]}

Eaten_Means_df = pd.DataFrame(Eaten_Means, columns = ['Shark', 'Mean'])

Eaten_Means_Sorted = Eaten_Means_df.sort_values(by=['Mean'], ascending = False)

fig = px.histogram(Eaten_Means_Sorted, x="Shark", y="Mean", 
                   title = "Mean Number of Fish Eaten by Shark",
                  labels={
                     "Shark": "Shark",
                     "Mean": " Mean Number of Fish",
                 })
fig.show()

In [51]:
# Mean of Ross and Chandler
Sandbar = eat['Ross'] + eat['Chandler']
Sandbar.mean()

6.439635535307517

In [52]:
# Mean of BT
BT = eat['BT1'] + eat['BT2'] + eat['BT3'] + eat['BT4'] + eat['BT5']
BT.mean()

13.128440366972477

In [53]:
# Mean of GR
GR = eat['GR1'] + eat['GR2'] + eat['GR3'] + eat['GR4'] + eat['GR5']
GR.mean()

7.892448512585812

In [77]:
# Graph Mean Number of Fish Eaten For Each Species

Species_Eaten_Means = {'Species': ['Blacktip', 'Gray Reef', 'Sandbar'],
        'Mean': [round(BT.mean(),3),round(GR.mean(),3), round(Sandbar.mean(),3)]}

Species_Eaten_Means_df = pd.DataFrame(Species_Eaten_Means, columns = ['Species', 'Mean'])

Species_Eaten_Means_Sorted = Species_Eaten_Means_df.sort_values(by=['Mean'], ascending = False)

fig = px.histogram(Species_Eaten_Means_Sorted, x="Species", y="Mean", 
                   title = "Mean Number of Fish Eaten by Species",
                  labels={
                     "Species": "Species",
                     "Mean": " Mean Number of Fish",
                 })
fig.show()

In [54]:
# Look at how many days the sharks ate in comparison to did not eat
columns = list(All_Sharks) 
  
for i in columns: 
    Days_Eaten = All_Sharks[i] > 0
    print(i, All_Sharks[i][Days_Eaten].count())

Ross 377
Chandler 337
BT1 324
BT2 355
BT3 331
BT4 367
BT5 370
GR1 282
GR2 286
GR3 307
GR4 318
GR5 313


In [55]:
# Proportion of Days Each Shark Ate Out of Total Days
columns = list(All_Sharks) 
  
for i in columns: 
    Days_Eaten = All_Sharks[i] > 0
    prop = round((All_Sharks[i][Days_Eaten].count())/All_Sharks[i].count(), 3)
    print(i, prop)

Ross 0.857
Chandler 0.768
BT1 0.74
BT2 0.811
BT3 0.756
BT4 0.838
BT5 0.849
GR1 0.645
GR2 0.654
GR3 0.703
GR4 0.728
GR5 0.716


In [73]:
# Graph Proportions for Each Shark
Eaten_Props = {'Shark': ['Ross', 'Chandler','BT1','BT2','BT3','BT4','BT5','GR1','GR2','GR3','GR4', 'GR5'],
        'Proportion': [0.857,0.768,0.74,0.811,0.756,0.838,0.849,0.645,0.654,0.703,0.728,0.716]}

Eaten_Props_df = pd.DataFrame(Eaten_Props, columns = ['Shark', 'Proportion'])

Eaten_Props_Sorted = Eaten_Props_df.sort_values(by=['Proportion'], ascending = False)

fig = px.histogram(Eaten_Props_Sorted, x="Shark", y="Proportion", 
                   title = "Proportion of Days Where The Shark Ate",
                  labels={
                     "Shark": "Shark",
                     "Proportion": "Proportion of Days",
                 })
fig.show()

# 
Out of all the sharks, Ross had the highest proportion of days eaten. He ate at least 1 fish for .857 of the days. The shark with the lowest proportion of days eaten was GR1, who ate at least 1 fish for .645 of the days.

In [60]:
# Look at the proportion of days eaten by each species
# Sandbars
SB_Days_Eaten = Sandbar > 0
SB_Eaten_Count = Sandbar[SB_Days_Eaten].count()
SB_Eaten_Prop = round(Sandbar[SB_Days_Eaten].count() / Sandbar.count(),3)
print("Sandbar Number of Days Eaten:", SB_Eaten_Count)
print("Sandbar Proportion of Days Eaten:", SB_Eaten_Prop)


Sandbar Number of Days Eaten: 424
Sandbar Proportion of Days Eaten: 0.966


In [61]:
# Blacktips
BT_Days_Eaten = BT > 0
BT_Eaten_Count = BT[BT_Days_Eaten].count()
BT_Eaten_Prop = round(BT[BT_Days_Eaten].count() / BT.count(),3)
print("Blacktip Number of Days Eaten:", BT_Eaten_Count)
print("Blacktip Proportion of Days Eaten:", BT_Eaten_Prop)

Blacktip Number of Days Eaten: 435
Blacktip Proportion of Days Eaten: 0.998


In [62]:
# Gray Reefs
GR_Days_Eaten = GR > 0
GR_Eaten_Count = GR[GR_Days_Eaten].count()
GR_Eaten_Prop = round(GR[GR_Days_Eaten].count() / GR.count(),3)
print("Gray Reef Number of Days Eaten:", GR_Eaten_Count)
print("Gray Reef Proportion of Days Eaten:", GR_Eaten_Prop)

Gray Reef Number of Days Eaten: 418
Gray Reef Proportion of Days Eaten: 0.957


# 
Blacktip sharks had the highest number of days where at least one shark from the group ate at least 1 fish. At least 1 Blacktip shark ate for .998 of the days. Gray Reef sharks only had at least one shark eat for .957 of the days, which was the lowest of the three groups

In [70]:
# Graph Proportions Based on Species of Shark
Species_Eaten_Props = {'Species': ['Blacktip','Sandbar','Gray Reef'],
        'Proportion': [BT_Eaten_Prop,SB_Eaten_Prop,GR_Eaten_Prop]
        }
Species_Eaten_Props_df = pd.DataFrame(Species_Eaten_Props, columns = ['Species', 'Proportion'])
fig = px.histogram(Species_Eaten_Props_df, x="Species", y="Proportion", 
                   title = "Proportion of Days Where A Species Ate",
                  labels={
                     "Species": "Shark Species",
                     "Proportion": "Proportion of Days",
                 })
fig.show()